core/templates/domain/statistics/playthrough-backend-api.service.spec.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for PlaythroughBackendApiService.
+ */
+
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+
+import { StorePlaythroughResponse, PlaythroughBackendApiService } from
+  'domain/statistics/playthrough-backend-api.service';
+import { PlaythroughObjectFactory } from
+  'domain/statistics/PlaythroughObjectFactory';
+
+describe('Playthrough backend api service', () => {
+  let pbas: PlaythroughBackendApiService;
```

Comment:
@kevinlee12 @bansalnitish have we made a style-decision on whether to use `this` for storing variables?

Reply:
Yep, I'm fine with using `this` for storing variables and making it a convention. But I would say, this can be done as part of a larger task to make all spec files use this convention. I feel introducing a new convention for just new spec files or files that are modified can lead to our codebase having different/conflicting patterns -- this can be confusing especially for new folks who write specs. Once we have a uniform pattern across all the spec files, we can have a linter to enforce the pattern for future PRs. WDYT? @brianrodri @bansalnitish @srijanreddy98